pkg/scheduler/framework/plugins/defaultpreemption/default_preemption_test.go

```diff
@@ -98,6 +101,10 @@ func getDefaultDefaultPreemptionArgs() *config.DefaultPreemptionArgs {
 	return dpa
 }
 
+var nodeResourcesFitFunc = func(plArgs apiruntime.Object, fh framework.Handle) (framework.Plugin, error) {
+	return noderesources.NewFit(plArgs, fh, utilfeature.DefaultFeatureGate.Enabled(features.PodOverhead))
```

Comment:
If we want to totally eliminate the dependency on `features`, we can just pass `true`. Or, it's tolerated to have this deps on tests.

Reply:
Ideally you need to test both, true and false, and not depend on the feature gate value at all.

Reply:
Prior to this, the tests already depended on the feature gate value anyway (since it was interpreted internally by the plugin)

Though, I think there are essentially tests for both cases (I noticed some specific tests around pod overhead that failed without the parameter, while most other tests didn't rely on it)

Reply:
For this particular case, whether the feature gate is enabled or not doesn't matter, the semantics is like "hey, I'm a consumer of NodeFit, I just want to derive whatever featuregate is set in your side".

Reply:
but Fit shouldn't know about feature gates. So this test should be either passing it from the feature gate or just pass whatever value you want.

I presume this test doesn't make use of PodOverhead, so it doesn't matter whether we enable the feature or not.

Reply:
I think we should completely cut the dependency of the plugins pkg on the features pkg. Where it makes sense, we should test both values, true and false. Perhaps in this specific case since it is not really testing the plugin in question it is ok to just select any.